library/core/src/num/nonzero.rs

```diff
@@ -287,6 +287,462 @@ nonzero_integers_div! {
     NonZeroUsize(usize);
 }
 
+// A bunch of methods for unsigned nonzero types only.
+macro_rules! nonzero_unsigned_operations {
+    ( $( $Ty: ident($Int: ty); )+ ) => {
+        $(
+            impl $Ty {
+                /// Add an unsigned integer to a non-zero value.
+                /// Return [`None`] on overflow.
+                ///
+                /// # Examples
+                ///
+                /// ```
+                /// #![feature(nonzero_ops)]
+                #[doc = concat!("# use std::num::", stringify!($Ty), ";")]
+                ///
+                #[doc = concat!("let one = ", stringify!($Ty), "::new(1).unwrap();")]
```

Comment:
This is completely a nit and not one that is applied much in the standard library, but when possible I find it better to promote use of `?` than `unwrap()` by wrapping the code in `/// # fn main() -> Option<()> {` and ending with `/// # Ok(()) }`.

This is optional though, and the examples are very good as-is !

Reply:
Alé, let's try this. But I guess you mean `-> Result<(), ()>`, right?

Reply:
Well, `Result<(), ()>` does not work either because `NonZeroUsize::new` returns an `Option` and not a `Result`. In consequence IIUC, the return type of the `main` function has to become `Result<(), std::option::NoneError>`, but then I need to decorate every example with `#[feature(try_trait)]`, and I am not sure whether it's a good idea that doctest for `#[feature(non_zero)]` depends on this other feature gate.. is that okay?

Reply:
The ? operator works with either Option or Result, as long as only one of them is used with ? inside the function

See https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator for more information

Reply:
Agreed, but this does not seem to work in the particular case of a `main` function

```
error[E0277]: `main` has invalid return type `Option<()>`
 --> src/main.rs:3:14
  |
3 | fn main() -> Option<()> {
  |              ^^^^^^^^^^ `main` can only return types that implement `Termination`
  |
  = help: consider using `()`, or a `Result`
```
.. thus the complications :\

Reply:
I always forget this trait 🤦 ! Sorry for that.

Then a (somewhat ugly) workaround is:

```rust
/// # fn inside() -> Option<()> {
/// ... code ...
/// # Ok(()); }
///
/// # fn main() { let _ = inside(); }
```

Again this is a nit, if you don't think it's that important feel free to ignore it !

Reply:
Haha, no worries. I went for a `# fn main() -> Result<(), ()>` with a hidden `# #![feature(try_trait)]`, let's see whether the other reviewers are happy with it :)